nix/module.nix

```
@@ -286,12 +359,16 @@ in
             expires 1d;
           '';
         in
-        if cfg.nginx != null then {
+        lib.mkIf (!cfg.useCaddy) {
           enable = lib.mkDefault true;
+          recommendedTlsSettings = cfg.useSSL;
+
           virtualHosts."oioioi" = {
             forceSSL = cfg.useSSL;
           } // cfg.nginx // {
             serverName = cfg.domain;
+            sslCertificateKey = lib.mkIf cfg.useSSL cfg.keyPath;
+            sslCertificate = lib.mkIf cfg.useSSL cfg.certPath;
```

```suggestion
} // (lib.optionalAttrs (cfg.useSSL && cfg.keyPath != null) {
            sslCertificateKey = cfg.keyPath;
}) //  (lib.optionalAttrs (cfg.useSSL && cfg.certPath != null) {
            sslCertificate = cfg.certPath;
}) // {
```
While the suggestion definitely doesn't look the prettiest there are issues with the way this is currently implemented:
1. lib.mkIf returns a special attribute set! This will cause these definitions to **always** override the values in cfg.nginx, even if useSSL is false. (doesn't *really* matter but is inconsistent with the way useSSL is handled above)
2. One may want to leave keyPath and certPath null to use ACME provided certificates, in this case useSSL will be true and a type error will occur because sslCertificate* options only accept path values. In this case, these options should remain unset here.